internal/util/setters/setters.go

```diff
@@ -96,48 +95,69 @@ func PerformSetters(path string) error {
 			continue
 		}
 
-		setter := &setters.PerformSetters{
-			Name:  fmt.Sprintf("gcloud.%s", c),
-			Value: v,
-			SetBy: "kpt",
-		}
-		rw := &kio.LocalPackageReadWriter{
-			PackagePath:           path,
-			KeepReaderAnnotations: false,
-			IncludeSubpackages:    true,
-		}
-		err = kio.Pipeline{
-			Inputs:  []kio.Reader{rw},
-			Filters: []kio.Filter{setter},
-			Outputs: []kio.Writer{rw},
-		}.Execute()
+		err = SetV1AutoSetter(fmt.Sprintf("gcloud.%s", c), v, path)
 		if err != nil {
 			return err
 		}
 
-		setter2 := &settersutil.FieldSetter{
-			Name:          fmt.Sprintf("gcloud.%s", c),
-			Value:         v,
-			SetBy:         "kpt",
-			OpenAPIPath:   filepath.Join(path, "Kptfile"),
-			ResourcesPath: path,
-		}
-		err = kio.Pipeline{
-			Inputs:  []kio.Reader{rw},
-			Filters: []kio.Filter{setter2},
-			Outputs: []kio.Writer{rw},
-		}.Execute()
+		err = SetV2AutoSetter(fmt.Sprintf("gcloud.%s", c), v, path)
 		if err != nil {
 			return err
 		}
+	}
 
-		if c == "core.project" && (setter.Count > 0 || setter2.Count > 0) {
-			// set the projectNumber if we set the projectID
-			projectID = v
-		}
+	return nil
+}
+
+var GetProjectNumberFromProjectID = func(projectID string) (string, error) {
+	gcloudCmd := exec.Command("gcloud",
+		"projects", "describe", projectID, "--format", "value(projectNumber)")
+	b, err := gcloudCmd.Output()
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get project number for %s, please verify gcloud "+
+			"credentials are valid and try again", projectID)
 	}
+	return strings.TrimSpace(string(b)), nil
+}
 
-	if projectID != "" {
+// DefExists returns true if the setterName exists in Kptfile definitions
+func DefExists(resourcePath, setterName string) bool {
+	if err := openapi.AddSchemaFromFile(filepath.Join(resourcePath, "Kptfile")); err != nil {
```

Comment:
The Kptfile name must be defined as a constant somewhere?

Reply:
Done.